legion/engine/physics/colliders/physicscollider.hpp

```diff
@@ -16,27 +16,27 @@ namespace legion::physics
     {
     public:
         bool shouldBeDrawn = true;
-        std::vector<std::unique_ptr<ConverganceIdentifier>> converganceIdentifiers;
+        std::vector<std::unique_ptr<ConvergenceIdentifier>> convergenceIdentifiers;
 
         PhysicsCollider()
         {
             static int colliderID = 0;
             id = colliderID++;
         }
 
-        virtual void AddConverganceIdentifier(const  physics_contact& contact) = 0;
+        virtual void AddConvergenceIdentifier(const  physics_contact& contact) = 0;
 
-        void AttemptFindAndCopyConverganceID(physics_contact& contact)
+        void AttemptFindAndCopyConvergenceID(physics_contact& contact)
         {
             if (!constants::applyWarmStarting) { return; }
 
-            for (auto&& converganceId : converganceIdentifiers)
+            for (auto&& convergenceId : convergenceIdentifiers)
             {
-                if (converganceId->refColliderID == contact.refCollider->GetColliderID())
+                if (convergenceId->refColliderID == contact.refCollider->GetColliderID())
                 {
-                    if (converganceId->IsEqual(contact))
+                    if (convergenceId->IsEqual(contact))
                     {
-                        converganceId->CopyLambdasToContact(contact);
+                        convergenceId->CopyLambdasToContact(contact);
```

Comment:
there is no move being done here, some of the floats of  Physics_Contact are being copied to ConvergenceIdentifier

Reply:
this this should still be an `auto&` since i don't recall calling functions or writing data to a rvalue reference being generally considered well defined. since rvalues are meant to be temporary values. (right hand side values)

Reply:
ah did not notice the '&&' will fix that